src/physics/utils/orbital_data.F90

```diff
@@ -0,0 +1,78 @@
+! Copyright (C) 2024 National Science Foundation-National Center for Atmospheric Research
+! SPDX-License-Identifier: Apache-2.0
+module orbital_data
+!--------------------------------------------------------------------------
+!
+! Provides access to conditions calculated based on the Earth's orbit.
+!
+!--------------------------------------------------------------------------
+
+  use shr_kind_mod,  only: kind_phys => SHR_KIND_R8
+  use shr_const_mod, only: PI => SHR_CONST_PI
+  use shr_orb_mod,   only: FILL_R8 => SHR_ORB_UNDEF_REAL
+
+  implicit none
+  private
+
+  public :: orbital_data_init, orbital_data_advance
+
+  !> \section arg_table_orbital_data Argument Table
+  !! \htmlinclude arg_table_orbital_data.html
+  !!
+  real(kind_phys),              protected, public :: solar_declination  = FILL_R8 ! Solar declination angle [radians]
+  real(kind_phys),              protected, public :: earth_sun_distance = FILL_R8 ! Earth-sun distance [AU]
+  real(kind_phys), allocatable, protected, public :: solar_zenith_angle(:)        ! Solar zenith angle (column) [radians]
+    
+!=======================================================================
+contains
+!=======================================================================
+
+  subroutine orbital_data_init(number_of_columns)
+
+    !-----------------------------------------------------------------------
+    !
+    ! Initialize the orbital data module.
+    !
+    !-----------------------------------------------------------------------
+
+    integer, intent(in) :: number_of_columns
+   
+    allocate(solar_zenith_angle(number_of_columns), source=FILL_R8)
```

Comment:
For these `allocate` calls (and elsewhere) could `check_allocate` (from `cam_abortutils`) be added after those as well?

Reply:
I think they are all updated now. Let me know if I missed any.